### ISSUE TITLE
[ENH] clarify and uniformize `model_path` parameter in foundation models, enable use of local weights and multiple hubs

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#7249]
We need to clarify and uniformize `model_path` parameters in the foundation models.

There are two problems we need to address:

1. Currently the docstrings are not clear what `model_path` can be. Even if the docstring says that it can be "string", it can surely not be any string. The docstrings must say what strings are admissible, and which condition needs to be satisfied, e.g., which model file needs to be in which huggingface repository, or on the local file system.

2. The ways in which the estimator accepts weights should be uniform across pre-trained models - e.g., both local checkpoint/weight files and huggingface should be possible. More generally, supporting other hubs like kaggle or replicate etc should be an aim.

It might be worth writing a "template docstring", and some common boilerplate for handling the most common checkpoint formats and hubs.

FYI @benHeid, @geetu040, @pranavvp16, @XinyuWuu - comments would be very much appreciated.

**Comment by benHeid** [comment]
I agree that the model_path should be consistent. Furthermore, we should think about if also a model directly can be passed. 

The string itself could always either show to a hf repo or local folder.

**Comment by fkiraly** [comment]
Yes, I think one should be able to pass a model directly as well, i.e., an in-memory model.

**Comment by fkiraly** [comment]
> The string itself could always either show to a hf repo or local folder.

How would the estimator know which of the two it refers to?

**Comment by benHeid** [comment]
> > The string itself could always either show to a hf repo or local folder.
> 
> 
> 
> How would the estimator know which of the two it refers to?

To be honest I am not sure how huggingface is resolving it. Probably check if it is local path otherwise fetch from the hub. It also uses caching i.e. if you already downloaded the model it will not download it again. (We should probably enable the user to set the force_download parameter)

**Comment by julian-fong** [comment]
From the `transformers` code base:

```python
pretrained_model_name_or_path (`str` or `os.PathLike`):
    Can be either:

        - A string, the *model id* of a pretrained model configuration hosted inside a model repo on
          huggingface.co.
        - A path to a *directory* containing a configuration file saved using the
          [`~PretrainedConfig.save_pretrained`] method, or the [`~PreTrainedModel.save_pretrained`] method,
          e.g., `./my_model_directory/`.
        - A path or url to a saved configuration JSON *file*, e.g.,
          `./my_model_directory/configuration.json`.
```

and some examples:

```python3
>>> from transformers import AutoConfig

>>> # Download configuration from huggingface.co and cache.
>>> config = AutoConfig.from_pretrained("google-bert/bert-base-uncased")

>>> # Download configuration from huggingface.co (user-uploaded) and cache.
>>> config = AutoConfig.from_pretrained("dbmdz/bert-base-german-cased")

>>> # If configuration file is in a directory (e.g., was saved using *save_pretrained('./test/saved_model/')*).
>>> config = AutoConfig.from_pretrained("./test/bert_saved_model/")

>>> # Load a specific configuration file.
>>> config = AutoConfig.from_pretrained("./test/bert_saved_model/my_configuration.json")

>>> # Change some config attributes when loading a pretrained config.
>>> config = AutoConfig.from_pretrained("google-bert/bert-base-uncased", output_attentions=True, foo=False)
>>> config.output_attentions
True

>>> config, unused_kwargs = AutoConfig.from_pretrained(
...     "google-bert/bert-base-uncased", output_attentions=True, foo=False, return_unused_kwargs=True
... )
>>> config.output_attentions
True
```

**Comment by julian-fong** [comment]
> We need to clarify and uniformize `model_path` parameters in the foundation models.
> 
> There are two problems we need to address:
> 
> 1. Currently the docstrings are not clear what `model_path` can be. Even if the docstring says that it can be "string", it can surely not be any string. The docstrings must say what strings are admissible, and which condition needs to be satisfied, e.g., which model file needs to be in which huggingface repository, or on the local file system.

The docstring I linked above can be a good starting point as to what 'types' are acceptable for the `model_path` parameter for huggingface, and then expand to any other hubs/resources we want to include. 

> 2. The ways in which the estimator accepts weights should be uniform across pre-trained models - e.g., both local checkpoint/weight files and huggingface should be possible. More generally, supporting other hubs like kaggle or replicate etc should be an aim.

Does this mean we would aim to have one singular method that handles all of model_loading/checkpoint_loading/weight_loading? and subsequently all implemented "hub-like" estimators would be required to use this function to instantiate their model in `fit`. 


I suppose that we would need to also ensure that there would be checks to ensure that the passed "string" is a valid one that makes sense for the given estimator. For example passing `/models/moirai_forecaster/` into the `ttm` `model_path` parameter.

**Comment by fkiraly** [comment]
My current opinion: let's just mirror precisely what huggingface does, I think the polymorphic single-parameter idea is the best:

![image](https://github.com/user-attachments/assets/0eee15c7-0cd2-45bd-9630-2e5d5854c249)

As a name, `model_path` is probably better.

**Comment by Spinachboul** [comment]
@fkiraly @benHeid 
I have opened a PR https://github.com/sktime/sktime/pull/7689/ which is related to this issue
I request you to kindly review the PR
thanks!!
